scalanet/src/io/iohk/scalanet/peergroup/PeerGroup.scala

```diff
@@ -144,10 +119,12 @@ object PeerGroup {
       * only to A or to the whole group that is referenced by the multicast address. Different
       * implementations could handle this logic differently.
       */
-    case class ChannelCreated[A, M](channel: Channel[A, M]) extends ServerEvent[A, M]
+    case class ChannelCreated[A, M](channel: Channel[A, M], release: Release) extends ServerEvent[A, M]
```

Comment:
Yes this is why we need this here. I know in the downstream project it would be less convenient to pass it around so we can keep wrapping it as it is now, but I think as you'll see in the next PR it's actually much easier this way to attach extra logic that needs to happen when a channel is closed.